packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/AnnotationParameterEnumValueSourceModel.java

```diff
@@ -0,0 +1,60 @@
+package dev.hilla.parser.models;
+
+import java.util.stream.Stream;
+
+import io.github.classgraph.AnnotationEnumValue;
+
+final class AnnotationParameterEnumValueSourceModel
+        extends AbstractModel<AnnotationEnumValue>
+        implements AnnotationParameterEnumValueModel {
+    private ClassInfoModel classInfo;
+
+    AnnotationParameterEnumValueSourceModel(AnnotationEnumValue origin) {
+        super(origin);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (!(obj instanceof AnnotationParameterEnumValueModel)) {
+            return false;
+        }
+
+        var other = (AnnotationParameterEnumValueModel) obj;
+
+        return getClassInfo().equals(other.getClassInfo())
+                && getValueName().equals(other.getValueName());
+    }
+
+    @Override
+    public ClassInfoModel getClassInfo() {
+        try {
+            if (classInfo == null) {
+                classInfo = ClassInfoModel
+                        .of(Class.forName(origin.getClassName()));
+            }
+
+            return classInfo;
+        } catch (ClassNotFoundException e) {
+            throw new ModelException(e);
```

Comment:
Checked exceptions rethrown as runtime exceptions: is this by design?

Reply:
Following discussion, it is OK since those exceptions shouldn't happen, so it is correct to report as RE if something happens here.